crates/red_knot_python_semantic/src/types/class_base.rs

```diff
@@ -54,9 +52,7 @@ impl<'db> ClassBase<'db> {
         KnownClass::Object
             .to_class_literal(db)
             .into_class_literal()
-            .map_or(Self::unknown(), |ClassLiteralType { class }| {
-                Self::Class(class)
-            })
+            .map_or(Self::unknown(), |literal| Self::Class(literal.class()))
```

Comment:
hrm, I preferred the code the way it was here -- I find unpacking the inner `Class` more readable when it comes to `ClassLiteralType` :/

it makes it easier for me to distinguish the concepts of a _class-literal type_ (a set that has exactly one inhabitant) and the underlying _class_ (the sole member of that set at runtime)

Reply:
That said, there probably would be benefits to treating the `class` field as fully opaque to external modules in the long term. And this probably gets us closer to that